### PR TITLE
Add block check when syncing block

### PR DESCRIPTION
**Pull request by NoelBright**
add TransactionCheck and HeaderCheck to blockchain.AddBlock.

### Type (put an `x` where ever applicable)
- [x] Bug fix: Link to the issue
- [ ] Feature (Non-breaking change)
- [ ] Feature (Breaking change)
- [ ] Documentation Improvement

### CODE REVIEW

File: block/block.go [hunk 1]
```diff
@@ -16,8 +16,10 @@ import (
 )
 
 type Block struct {
-	Header       *Header
-	Transactions []*transaction.Transaction
+	Header                *Header
+	Transactions          []*transaction.Transaction
+	IsHeaderChecked       bool
+	IsTransactionsChecked bool
```
Comment (gdmmx) [inline block/block.go]: How about rename `IsTransactionsChecked` as `IsTxnsChecked`?

File: chain/blockchain.go [hunk 2]
```diff
@@ -57,6 +57,7 @@ func (bc *Blockchain) AddBlock(block *block.Block, fastAdd bool) error {
 
 	err := bc.SaveBlock(block, fastAdd)
 	if err != nil {
+		log.Error("AddBlock error, ", err)
```
Comment (gdmmx) [inline chain/blockchain.go]: Why `SaveBlock` error but log output "**AddBlock** error"?

File: chain/blockchain.go [hunk 3]
```diff
@@ -72,6 +73,24 @@ func (bc *Blockchain) GetHeader(hash Uint256) (*block.Header, error) {
...
+			}
+			block.IsHeaderChecked = true
+		}
+
+		if !block.IsTransactionsChecked {
+			err := TransactionCheck(context.Background(), block)
+			if err != nil {
+				return err
+			}
+			block.IsTransactionsChecked = true
```
Comment (gdmmx) [inline chain/blockchain.go]: Same as above. Short name for srcCode concision.

File: chain/blockchain.go (hunk 2 above)
Comment (NoelBright) [reply chain/blockchain.go]: because this statement is in AddBlock function.

File: block/block.go (hunk 1 above)
Comment (NoelBright) [reply block/block.go]: done

File: chain/blockchain.go (hunk 3 above)
Comment (NoelBright) [reply chain/blockchain.go]: done

**Pull request merged**